service/frontend/workflow_handler_deprecated.go

```diff
@@ -182,7 +182,8 @@ func (wh *WorkflowHandler) getWorkflowExecutionHistory(
 					ctx,
 					wh.metricsScope(ctx),
 					namespaceID,
-					*execution,
+					// FIXME shallow copy
+					common.CloneProto(execution),
```

Comment:
1. WorkflowExecution has only simple fields, so this already is a shallow copy
2. I would do it like this: just pass the pointer here. then on the four lines that set execution.RunId, have those lines do the copy like `execution = &commonpb.WorkflowExecution{WorkflowId: execution.WorkflowId, RunId: newRunId}`. that way we treat WorkflowExecution as immutable.

Reply:
I can't actually figure out why we shallow copied these in the first place; I don't see any mutations in these calls